### PR TITLE
feat: added "orderByNeedles" option

### PR DESCRIPTION
**Pull request by simlu**

### CODE REVIEW

File: src/core/find.js [hunk 1]
```diff
@@ -186,7 +187,16 @@ module.exports = (haystack_, searches_, ctx) => {
             }
           }
         }
-        stack.push(false, searchesOut, isArray ? Number(key) : key, depth + 1);
+        if (ctx.orderByNeedles) {
+          searchesOut.index = Buffer.from(searchesOut.map((e) => getOrder(e)).sort());
```
Comment (simlu) [inline src/core/find.js:191]:
```suggestion
          searchesOut.index = Buffer.from(searchesOut.map((e) => getOrder(e)).sort());
```
why do we need to sort this?

Comment (simlu) [reply src/core/find.js:191]: Because different searches are pursued in parallel

File: test/index.spec/order-by-needles.spec.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    const r = exec(['b', '', 'a'], { a: 0, b: 1 });
+    expect(r).to.deep.equal(['b', 'a', '']);
+  });
+
+  it('Testing edge case ordering', () => {
+    const r = exec(['b.c', 'a', 'b', 'd'], { a: 0, b: { c: 1 }, d: 2 });
+    expect(r).to.deep.equal([
+      'b.c', 'b', 'a', 'd'
+    ]);
+  });
```
Comment (simlu) [inline test/index.spec/order-by-needles.spec.js:93]: Todo: add test to ensure that order for starstar is always consistent

Comment (simlu) [reply test/index.spec/order-by-needles.spec.js:93]: After merge add regression test

**Comment by simlu** [comment]
This pr introduces a minor regression in compile time.

**Should setOrder only be called if orderByNeedles is true?**
